### PR TITLE
Basemap and labels icons

### PR DESCRIPTION
**Pull request by alonsogarciapablo**
Part of https://github.com/CartoDB/cartodb/issues/7342. This PR implements some changes to make layer icons better. Here're some examples:

**Basemap with labels on top:**

screen shot 2016-06-17 at 17 29 32

**Color base layer:**

screen shot 2016-06-17 at 17 30 14

**Background image base layer:**

screen shot 2016-06-17 at 17 34 10

@piensaenpixel We need an icon for WMS layers. I left a TODO that will be resolved before merging this PR.

@viddo apart from the missing icon ^^, PR should be ready. I'd love to hear your thoughts and feedback about the code 😍

### CODE REVIEW

File: lib/assets/javascripts/cartodb3/data/layer-definitions-collection.js [hunk 1]
```diff
@@ -375,33 +375,15 @@ module.exports = Backbone.Collection.extend({
   },
 
   _createBaseLayer: function (newBaseLayerAttrs) {
-    var optionsAttribute = {};
```
Comment (alonsogarciapablo) [inline lib/assets/javascripts/cartodb3/data/layer-definitions-collection.js]: I removed these TODOs (not related with this PR).

cc: @matallo @viddo

Comment (viddo) [inline lib/assets/javascripts/cartodb3/data/layer-definitions-collection.js]: 👍

Comment (matallo) [inline lib/assets/javascripts/cartodb3/data/layer-definitions-collection.js]: roger

File: lib/assets/javascripts/cartodb3/editor/layers/layer-views/data-layer-view.js [hunk 2]
```diff
@@ -59,9 +56,10 @@ module.exports = CoreView.extend({
     }
 
     if (m.get('source')) {
-      var view = this._newAnalysesView(this.$('.js-analyses'), m);
-      this.addView(view);
-      view.render();
+      var analysesView = this._newAnalysesView(this.$('.js-analyses'), m);
+      analysesView.bind('nodeClicked', this._onNodeClicked, this);
```
Comment (viddo) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-views/data-layer-view.js]: This looks wrong, I don't think this event/method exist anymore

File: lib/assets/javascripts/cartodb3/editor/layers/layer-views/data-layer-view.js [hunk 3]
```diff
@@ -16,14 +12,15 @@ module.exports = CoreView.extend({
...
+    'click .js-thumbnail': '_onEditLayer',
     'click .js-title': '_onEditLayer',
     'click .js-toggle': '_onToggleLayerClicked'
   },
 
   initialize: function (opts) {
     if (!opts.stackLayoutModel) throw new Error('stackLayoutModel is required');
     if (!opts.layerDefinitionsCollection) throw new Error('layerDefinitionsCollection is required');
-    if (!_.isFunction(opts.newAnalysesView)) throw new Error('newAnalysesView is required as a function');
+    if (!opts.newAnalysesView) throw new Error('newAnalysesView is required');
```
Comment (viddo) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-views/data-layer-view.js]: Why relax this check?

**Comment by viddo** [comment]
Some minor details, overall 👍

File: lib/assets/javascripts/cartodb3/editor/layers/layer-views/data-layer-view.js (hunk 2 above)
Comment (alonsogarciapablo) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-views/data-layer-view.js]: Good catch. I renamed `view` and then I didn't spot this line when solving some merge conflicts.

**Comment by saleiva** [comment]
yay!

**Comment by alonsogarciapablo** [comment]
Thanks @viddo. Feeback addresed.

@piensaenpixel Added the `is-hidden` classes to data layers in https://github.com/CartoDB/cartodb/pull/7928/commits/78d04651275cdb5a5c1c68c77cbaf0b0ee04c37d.

We just need the icon for WMS layers now and this should be ready to 🚀

**Pull request merged**